### PR TITLE
Implemented support to store user preference for selected product store (#2f2h8hu)

### PR DESCRIPTION
**Pull request by rathoreprashant**
### Related Issues

Closes #

### Short Description and Why It's Useful

Added user preference for selected product store

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/receiving#contribution-guideline)

### CODE REVIEW

File: src/store/modules/user/actions.ts [hunk 1]
```diff
@@ -52,22 +52,37 @@ const actions: ActionTree<UserState, RootState> = {
...
+  async getProfile ( { commit, dispatch }) {
     const resp = await UserService.getProfile()
     if (resp.status === 200) {
       const localTimeZone = moment.tz.guess();
       if (resp.data.userTimeZone !== localTimeZone) {
         emitter.emit('timeZoneDifferent', { profileTimeZone: resp.data.userTimeZone, localTimeZone});
       }
+
       commit(types.USER_INFO_UPDATED, resp.data);
+      await dispatch('getEComStores', { facilityId: resp.data.facilities[0] })
```
Comment (disha1202) [inline src/store/modules/user/actions.ts:64]: Check for facilities in resp before dispatching getEComStores.

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts [hunk 2]
```diff
@@ -87,6 +102,45 @@ const actions: ActionTree<UserState, RootState> = {
...
+          "storeName_op": "not-empty"
+        },
+        "fieldList": ["productStoreId", "storeName"],
+        "entityName": "ProductStore",
+        "distinct": "Y",
+        "noConditionFind": "Y"
+      }
+
+      resp = await UserService.getEComStores(param);
+      if(resp.status === 200 && resp.data.docs?.length > 0 && !hasError(resp)) {
```
[the remark this answers is not in the document]
Comment (rathoreprashant) [reply src/store/modules/user/actions.ts]: Updated ma'am

File: src/views/Settings.vue [hunk 3]
```diff
@@ -9,25 +9,32 @@
...
-       <ion-item>
-         <ion-icon :icon="storefrontOutline" slot="start" />
-         <ion-label>{{$t("Store")}}</ion-label>
-         <ion-select interface="popover" :placeholder="$t('store name')" :value="currentFacility.facilityId" @ionChange="setFacility($event)">
-           <ion-select-option v-for="facility in (userProfile && userProfile.facilities ? userProfile.facilities : [])" :key="facility.facilityId" :value="facility.facilityId" >{{ facility.name }}</ion-select-option>
-         </ion-select>
-       </ion-item>
+        <ion-item>
+          <ion-icon :icon="globeOutline" slot="start" />
+          <ion-label>{{$t("eCom Store")}}</ion-label>
```
[the remark this answers is not in the document]
Comment (rathoreprashant) [reply src/views/Settings.vue]: improved

File: src/views/Settings.vue [hunk 4]
```diff
@@ -9,25 +9,32 @@
...
+          <ion-select interface="popover" :value="currentEComStore.productStoreId" @ionChange="setEComStore($event)">
+            <ion-select-option v-for="store in (userProfile ? userProfile.stores : [])" :key="store.productStoreId" :value="store.productStoreId" >{{ store.storeName }}</ion-select-option>
+          </ion-select>
+        </ion-item>
+        <ion-item>
+          <ion-icon :icon="storefrontOutline" slot="start" />
+          <ion-label>{{$t("Store")}}</ion-label>
```
[the remark this answers is not in the document]
Comment (rathoreprashant) [reply src/views/Settings.vue]: improved

File: src/store/modules/user/actions.ts (hunk 1 above)
Comment (rathoreprashant) [reply src/store/modules/user/actions.ts]: Updated ma'am

**Pull request merged**